### DISCUSSION TITLE
SX1280 - Transmitter / Receiver - Recommended config

### DISCUSSION THREAD

**Discussion by alx-uta** [jgromes/RadioLib#921] [General]
First of all, I'd like to say thank you for this great library!

I'm currently working on a custom RC Transmitter and I'm using FSK to send and receive the data.

Main Hardware:
- ESP32-WROVER
- SX1280

Current settings:
```
float   SX1280_FREQUENCY            = 2410;
int     SX1280_FREQUENCY_DEVIATION  = 200;
int     SX1280_BIT_RATE             = 250;
int     SX1280_OUTPUT_POWER         = 8; // -18 to 13 dBm
int     SX1280_GAIN_CONTROL         = 8; // 1 - 13
uint8_t SX1280_DATA_SHAPING         = RADIOLIB_SHAPING_0_5; // (0x02)
int     SX1280_CRC_VALUE            = 2;
int     SX1280_PREAMBLE_LENGTH      = 4;
```

Extra variable for the RX.
```
bool    SX1280_SET_HIGH_SENSITIVITY_MODE    = true;
```
Payload size: 4 - 20 bytes.

Is there something that I should consider to have a good balance between speed and range?

Thanks

**Comment by jgromes** [comment]
> Is there something that I should consider to have a good balance between speed and range?

If power consumption is not a (significant) factor, then you should be able to increase transmitter power to 13 dBm. I would also recommend leaving gain control automatic. Other than that, you are using one of the lower bit rate settings which will improve the range. Though just by using a 2.4 GHz module, it is going to be impacted, just because higher frequencies get attenuated faster than sub-GHz. But overall, it's hard to judge without knowing what are the actual range/speed/power consumption requirements.

**Reply by alx-uta** [reply]
Thank you for the reply!
I'm not that concerned about the power consumption. Currently, I'm using a 21700/4200mAh battery and it lasts quite a long time.

Sounds good, I'll set the power to 13 dBm and I'll remove the gain control part.

Are there any other special settings that I should have a look at? I checked the documentation from https://jgromes.github.io/RadioLib/class_s_x128x.html and there's nothing I could think of.

For the TX I can see that we have .setOutputPower(). Is there something similar for the RX? Or, this is mainly based on what antenna is used?

**Reply by jgromes** [reply]
> For the TX I can see that we have .setOutputPower(). Is there something similar for the RX?

I guess that would be the gain, but I would really recommend to keep it automatic, unless you have the equipment to evaluate which setting is the best for your specific use case and hardware. And yes, the antenna quality will have a massive impact. It pays to check if the antenna is actually tuned well - out of experience some vendors will happily sell sub-GHz antenna marked as 868 MHz that is actually tuned to something like 750 MHz.

**Reply by alx-uta** [reply]
Sounds good. I'll keep this in mind and run some more tests.

Thanks
